src/util/memory.rs

```diff
@@ -13,8 +13,12 @@ pub fn result_is_mapped(result: Result<()>) -> bool {
 }
 
 pub fn zero(start: Address, len: usize) {
+    set(start, 0, len);
+}
+
+pub fn set(start: Address, val: u8, len: usize) {
     let ptr = start.to_mut_ptr();
-    wrap_libc_call(&|| unsafe { libc::memset(ptr, 0, len) }, ptr).unwrap()
+    wrap_libc_call(&|| unsafe { libc::memset(ptr, val as i32, len) }, ptr).unwrap()
```

Comment:
Can we use `std::ptr::write_bytes<u8>` instead?

It seems the underlying LLVM compiler can treat `libc::memset` as an intrinsic function, but I am not sure whether it always behaves this way.

Reply:
What are the concerns of using `memset`?

Reply:
It just feels strange to me that we have a function in the Rust standard library for this, but we used something in libc instead.

Reply:
Performance-wise, I wrote a microbenchmark that sets the contents of a byte array.  There is no performance difference between `write_bytes` and `memset` no matter whether I write the whole array with one `write_bytes`/`memset` or write each element individually in a loop using those functions.

Reply:
> It just feels strange to me that we have a function in the Rust standard library for this, but we used something in libc instead.

Sounds reasonable. We can change it to the equivalent in Rust standard library. I guess we knew we wanted `memset`, and we could access it from `libc`. So we did not try to find if there is an equivalent in the Rust library

Reply:
Done. `memory::set` is now implemented with `ptr::write_bytes<u8>`.